src/main/kotlin/org/rust/lang/core/psi/RustItemElement.kt

```diff
@@ -1,26 +1,15 @@
 package org.rust.lang.core.psi
 
-import org.rust.lang.core.symbols.RustQualifiedPath
-import org.rust.lang.core.symbols.RustQualifiedPathPart
+import org.rust.lang.core.symbols.RustPath
 
 interface RustItemElement : RustVisibilityOwner, RustOuterAttributeOwner
 
-/**
- * Returns a fully-qualified [RustQualifiedPath] representing a 'path' to this [RustItemElement]
- * from the crate root, like `foo::bar::baz`.
- *
- * This path is NOT guaranteed to be unique: items from different crates
- * can have the same path within respective crates.
- */
-val RustItemElement.canonicalCratePath: RustQualifiedPath
-    get() =
-        if (this is RustMod)
-            @Suppress("USELESS_CAST")
-            (this as RustMod).canonicalCratePath!!
-        else
-            RustQualifiedPath.create(
-                RustQualifiedPathPart.from(name),
-                qualifier = containingMod?.canonicalCratePath,
-                fullyQualified = true
-            )
+val RustItemElement.canonicalCratePath: RustPath? get() {
+    if (this is RustMod) {
+        @Suppress("USELESS_CAST")
+        return (this as RustMod).canonicalCratePath
```

Comment:
While technically it is not a useless cast, I kind'a like this warning, because defining two extension methods for non disjoint receiver types is scary. 

I think we need some better abstraction here: `ForeignMod` is an item, but it can't be identified with a path, while `EnumVariant` is not an item, but it has a reasonable `path`. And there are also things like

```
fn foo() {
    { struct S; }
    { struct S; }
}
```